plugins/postcss-cascade-layers/src/index.ts

```diff
@@ -19,11 +20,15 @@ function postcssCascadeLayers(opts) {
 
 				// check for where a layer has nested layers AND styles outside of those layers
 				atRule.each((node) => {
-					if (node.type == "atrule") {
+					if (node.type == 'atrule' && node.name === 'layer') {
 						hasNestedLayers = true;
-					} else if (node.type == "rule") {
+					} else {
```

Comment:
This actually needs much more complex logic, but is ok for now.
Needs to traverse the entire child tree.

Reply:
I can also be totally wrong about this.
I suspect there can be edge cases with `@keyframes`, `@supports`, `@media`, ...
Easier to handle this once it comes together a bit and we can test things in a browser.